### PR TITLE
Idea 288 - ranking badges

### PR DESCRIPTION
**Pull request by tehandyb**

### CODE REVIEW

File: src/util/NumberRank.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      // If the previous value equals the current value, the current value is tied/set to the same rank as the previous value
+      if(sortedValues[i-1] === sortedValues[i]) {
+        rankArray.push(rankArray[i-1])
+      }
+      else {
+        rankArray.push(i+1)
+      }
+    }
+    else { // Start with rank 1 for the first element
+      rankArray.push(1)
```
Comment (AlwaysMoreHats) [inline src/util/NumberRank.js]: There's a trick you can do here to reduce nesting. Before the for loop, add `rankArray.push(1)` and in the for loop, it should look like `for(var i = 0; i < sortedValues.length; i ++)`. I'm actually surprised that line 10 doesn't break.

**Review by AlwaysMoreHats — COMMENTED** [review]
My only comment is fine if everything works. What I"m requesting is a "refactoring" change instead of functional

File: src/util/NumberRank.js (hunk 1 above)
Comment (tehandyb) [reply src/util/NumberRank.js]: That for loop is the same as the one above, do you mean i = 1 instead of i = 0?

Comment (AlwaysMoreHats) [reply src/util/NumberRank.js]: Yes, my bad

**Review by jackcompton — APPROVED** [review]

File: src/util/NumberRank.js (hunk 1 above)
Comment (tehandyb) [reply src/util/NumberRank.js]: Ok, fixed that up.

**Pull request merged**